File: java/admob/FullScreenNativeExample/app/src/main/java/com/google/example/gms/fullscreennativeexample/MyApplication.java
package com.google.example.gms.fullscreennativeexample;

import android.app.Activity;
import android.app.Application;
import android.content.Context;
import androidx.annotation.NonNull;
import com.google.android.gms.ads.AdListener;
import com.google.android.gms.ads.AdLoader;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.LoadAdError;
import com.google.android.gms.ads.nativead.NativeAd;

/** Application class that loads a native ad.. */
public class MyApplication extends Application {
  private NativeAdManager nativeAdManager;

  @Override
  public void onCreate() {
    super.onCreate();

    nativeAdManager = new NativeAdManager();
  }

  public NativeAd getNativeAd() {
    return nativeAdManager.getNativeAd();
  }

  /**
   * Load an app open ad.
   *
   * @param activity the activity that shows the app open ad
   */
  public void loadAd(
      @NonNull Activity activity,
      @NonNull NativeAdEventListener nativeAdEventListener) {
    // We wrap the loadAd to enforce that other classes only interact with MyApplication
    // class.
    nativeAdManager.loadAd(activity, nativeAdEventListener);
  }

  /**
   * Interface definition for a callback to be invoked when an app open ad is complete (i.e.
   * dismissed or fails to show).
   */
  public interface NativeAdEventListener {
    void onNativeAdLoaded(boolean success);
  }

  /** Inner class that loads and shows native ads. */
  private class NativeAdManager {
    private static final String AD_UNIT_ID = "ca-app-pub-3940256099942544/2247696110";
    private NativeAd nativeAd;

    public NativeAd getNativeAd() {
      return nativeAd;
    }

    /** Constructor. */
    public NativeAdManager() {}

    /**
     * Creates a request for a new native ad based on the boolean parameters and calls the
     * corresponding "populate" method when one is successfully returned.
     */
    public void loadAd(Context context, @NonNull NativeAdEventListener nativeAdEventListener) {
      AdLoader adLoader =
          new AdLoader.Builder(context, AD_UNIT_ID)
              .forNativeAd(nativeAd -> {
                this.nativeAd = nativeAd;
                nativeAdEventListener.onNativeAdLoaded(true);
              })
              .withAdListener(new AdListener() {
                @Override
                public void onAdFailedToLoad(@NonNull LoadAdError adError) {
                  nativeAdEventListener.onNativeAdLoaded(false);
                }
              })
              .build();

      adLoader.loadAd(new AdRequest.Builder().build());
    }
  }
}

File: kotlin/admanager/AppOpenExample/app/src/main/java/com/google/android/gms/example/appopenexample/MyApplication.kt
package com.google.android.gms.example.appopenexample

import android.app.Activity
import android.app.Application
import android.os.Bundle
import androidx.multidex.MultiDexApplication

/** Application class that initializes, loads and show ads when activities change states. */
// [START application_class]
class MyApplication : MultiDexApplication(), Application.ActivityLifecycleCallbacks {

  private lateinit var appOpenAdManager: AppOpenAdManager

  override fun onCreate() {
    super<MultiDexApplication>.onCreate()
    registerActivityLifecycleCallbacks(this)

    appOpenAdManager = AppOpenAdManager(this)
    appOpenAdManager.initialize()
  }

  // [END application_class]

  /** ActivityLifecycleCallback methods. */
  // [START activity_lifecycle_callbacks]
  override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {}

  override fun onActivityStarted(activity: Activity) {
    // An ad activity is started when an ad is showing, which could be AdActivity class from Google
    // SDK or another activity class implemented by a third party mediation partner. Updating the
    // currentActivity only when an ad is not showing will ensure it is not an ad activity, but the
    // one that shows the ad.
    if (!appOpenAdManager.isShowingAd) {
      appOpenAdManager.currentActivity = activity
    }
  }

  override fun onActivityResumed(activity: Activity) {}

  override fun onActivityPaused(activity: Activity) {}

  override fun onActivityStopped(activity: Activity) {}

  override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}

  override fun onActivityDestroyed(activity: Activity) {}

  // [END activity_lifecycle_callbacks]

  /**
   * Shows an app open ad.
   *
   * @param activity the activity that shows the app open ad
   * @param onShowAdComplete the function to be notified when an app open ad is complete
   */
  fun showAdIfAvailable(activity: Activity, onShowAdComplete: () -> Unit) {
    // We wrap the showAdIfAvailable to enforce that other classes only interact with MyApplication
    // class.
    appOpenAdManager.showAdIfAvailable(activity, onShowAdComplete)
  }

  /**
   * Load an app open ad.
   *
   * @param activity the activity that shows the app open ad
   */
  fun loadAd(activity: Activity) {
    // We wrap the loadAd to enforce that other classes only interact with MyApplication
    // class.
    appOpenAdManager.loadAd(activity)
  }
}
